include/deal.II/base/quadrature_lib.h

```diff
@@ -647,6 +647,205 @@ class QGaussLobattoChebyshev : public Quadrature<dim>
 
 };
 
+/**
+ * Given an arbitrary quadrature formula, return one that chops the quadrature
+ * points above the hyper-plane defined by $\sum_i x_i = 1$. In other words,
+ * it extracts those quadrature points from the base formula that satisfy
+ * $\sum_i (\mathbf x_q)_i \le 1$."
+ *
+ * In general the resulting quadrature is not very useful, unless the
+ * quadrature you started from has been constructed specifically to integrate
+ * over triangles or tetrahedra. This class ensures that the resulting
+ * quadrature formula only has quadrature points in the reference simplex or on
+ * its boundary.
+ *
+ * No transformation is applied to the weights. The weights referring to points
+ * that live outside the reference simplex are simply discarded. If you chop a
+ * quadrature formula that is good for both the lower and upper triangles
+ * separately, then the weights should add up to 1/2.
```

Comment:
You probably want to say "then the weights should add up to 1/2 for each of the two formulas".

By the way, don't all useful quad quadrature formulas have points that are in fact *on* the diagonal? Wouldn't that make them bad candidates as base quadratures?

Reply:
Actually, not really. Most useful quadrature formulas are derived from Gauss quads, without end points, and then through a transformation to a reference triangle (for example: Duffy and TrianglePolar). If the base quadrature has no points on the boundary, then the resulting quadrature has no points on the diagonal.